Content.Shared/Rounding/ContentHelpers.cs

```diff
@@ -110,7 +110,11 @@ public static int RoundToEqualLevels(double actual, double max, int levels)
                 return 0;
             }
 
-            return (int) Math.Round(actual / max * levels, MidpointRounding.ToZero);
+            // This is necessary if the stack size is the same as the number of levels.
+            // If this doesn't happen, it starts at 1 instead of 0 and will skip the first state.
+            var mod = levels == (int) max ? 1 : 0;
```

Comment:
int to double comparison is suspect at best, even with the int casting. Also, how is this meant to handle noninteger values for `max`? If you actually want it to do a Math.Floor transformation then you should do that specifically instead of what appears to be kinda suspect logic.